### PR TITLE
org.radare.Cutter

### PR DESCRIPTION
**Pull request by jcaesar**
Hey @chergert 

Likely still needs some discussion, e.g. about `--filesystem=host`.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.radare.Cutter

File: org.radare.Cutter.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      path: fix-capstone.patch
+    - type: patch
+      path: fix-zip.patch
+    - type: patch
+      path: fix-lib-dist.patch
+    - type: git
+      url:  https://github.com/aquynh/capstone.git
+      commit: 37569a6874c8547b349a80823adda9284499fe80
+      dest: shlr/capstone
+   config-opts:
```
Comment (barthalion) [inline org.radare.Cutter.yml]: Nitpick: we usually put `sources` last.

File: org.radare.Cutter.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - --with-syscapstone
+    - --with-syszip
+    - --with-openssl
+
+ - name: cutter
+   sources:
```
Comment (barthalion) [inline org.radare.Cutter.yml]: Ditto.

File: org.radare.Cutter.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+id: org.radare.Cutter
```
Comment (barthalion) [inline org.radare.Cutter.yml]: File needs some adjustments to follow our YAML style guide: https://github.com/flathub/flathub/wiki/YAML-Style-Guide

File: org.radare.Cutter.yml (hunk 1 above)
Comment (jcaesar) [reply org.radare.Cutter.yml]: I've seen that, but always found it strange. You first download stuff, and then you build it. May I ask why?

File: org.radare.Cutter.yml (hunk 3 above)
Comment (jcaesar) [reply org.radare.Cutter.yml]: ✓

**Comment by jcaesar** [comment]
Btw, I should mention that I have no affiliation with radare and am building this as per @chergert's request.

**Comment by jcaesar** [comment]
Please do not accept this yet, I just learned that this needs a jupyter install for one of its menus.

**Comment by TingPing** [comment]
bot, build

**Comment by TingPing** [comment]
bot, build org.radare.Cutter

File: org.radare.Cutter.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    sources:
+      - type: archive
+        url: https://github.com/radareorg/cutter/archive/v1.4.tar.gz
+        sha256: f2fff960ea102fc8cf7d62fe25ee2c992b2e19857a4f6a2e308b3c9cba93e5b8
+      - type: file
+        path: org.radare.Cutter.appdata.xml
+
+  - jupyter.json
+    # evil magic to get the pip install sources (there has to be an easier way…):
+    # flatpak run --share=network --command=python3 org.kde.Sdk//5.11 -m pip install jupyter 2>/dev/null | sed -nur 's/^.*(http[^ ]*).*$/\1/p' | while read l; do ( mkdir dl; cd dl; wget -q "$l"; sha=$(sha256sum *| cut -d\  -f1); cd ..; mkdir -p .flatpak-builder/downloads/; mv dl .flatpak-builder/downloads/$sha; echo '{ "type": "file", "sha256": "'"$sha\", \"url\": \"$l\" }" ) || rm -rf dl; done
```
Comment (barthalion) [inline org.radare.Cutter.yml]: The easier way is using pip generator from [here](https://github.com/flatpak/flatpak-builder-tools/tree/master/pip).

**Comment by barthalion** [comment]
bot, build org.radare.Cutter

**Comment by barthalion** [comment]
Have you contacted upstream about adding their application to Flathub?

**Comment by jcaesar** [comment]
I have. They took the appstream xml and gave me some tips, but they're pretty set on appimage distribution, so I deem it unlikely that there will be strong cooperation.

**Comment by jcaesar** [comment]
I'm sorry, this is blocked because I don't understand what to do about this error https://flathub.org/builds/#/builders/32/builds/268/steps/6/logs/stdio

**Comment by barthalion** [comment]
bot, build org.radare.Cutter

File: org.radare.Cutter.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>org.radare.Cutter.desktop</id>
```
Comment (barthalion) [inline org.radare.Cutter.appdata.xml]: Should `.desktop` be here? Also should appdata provide <launchable>? Trying to figure out CI failure.

File: org.radare.Cutter.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+id: org.radare.Cutter
+runtime: org.kde.Platform
+runtime-version: 5.11
+sdk: org.kde.Sdk
+command: Cutter
+rename-desktop-file: Cutter.desktop
+rename-icon: cutter
+rename-appdata-file: Cutter.appdata.xml
```
Comment (barthalion) [inline org.radare.Cutter.yml]: You are installing it manually below using $FLATPAK_ID var so rename can be removed.

**Comment by TingPing** [comment]
bot, build org.radare.Cutter

**Comment by barthalion** [comment]
bot, build org.radare.Cutter

**Comment by jcaesar** [comment]
I think there's some crash with keypresses still, so I'll have to give it some more work anyway.

Or is it just my machine being weird again and you don't see
```
ASSERT: "uint(i) < uint(size())" in file ../../../../include/QtCore/../../src/corelib/tools/qstring.h, line 931
```
when you try to type something into any input field?

**Comment by barthalion** [comment]
I'm closing it due to inactivity – but please re-open it or resubmit as new pull request if you would like to see it merged in the future.

**Pull request closed without merging**